Mathlib/Topology/Order/HullKernel.lean

```diff
@@ -0,0 +1,250 @@
+`hull` respectively. They form an antitone Galois connection between the subsets of `T` and `α`.
+When `α` can be generated from `T` by taking infs, this becomes a Galois insertion and the relative
+topology (`Topology.lower`) on `T` takes on a particularly simple form: the relative-open sets are
+exactly the sets of the form `(hull T a)ᶜ` for some `a` in `α`. The topological closure coincides
+with the closure arising from the Galois insertion. For this reason the relative lower topology on
+`T` is often referred to as the "hull-kernel topology". The names "Jacobson topology" and "structure
+topology" also occur in the literature.
+
+## Main statements
+
+- `PrimitiveSpectrum.relativeLowerIsTopologicalBasis` - the sets `(hull a)ᶜ` form a basis for the
+  relative lower topology on `T`.
+- `PrimitiveSpectrum.isOpen_iff` - for a complete lattice, the sets `(hull a)ᶜ` are the relative
+  topology.
+- `PrimitiveSpectrum.gc` - the `kernel` and the `hull` form a Galois connection
+- `PrimitiveSpectrum.gi` - when `T` generates `α`, the Galois connection becomes an insertion.
+- `PrimitiveSpectrum.lowerTopology_closureOperator_eq_gc_closureOperator` - relative lower topology
+  closure coincides with the "hull-kernel" closure arising from the Galois insertion.
+
+## Implementation notes
+
+The antitone Galois connection from `Set T` to `α` is implemented as a monotone Galois connection
+between `Set T` to `αᵒᵈ`.
+
+## Motivation
+
+The motivating example for the study of a set `T` of prime elements which generate `α` is the
+primitive spectrum of the lattice of M-ideals of a Banach space.
+
+## References
+
+* [Gierz et al, *A Compendium of Continuous Lattices*][GierzEtAl1980]
+* [Henriksen et al, *Joincompact spaces, continuous lattices and C*-algebras*][henriksen_et_al1997]
+
+## Tags
+
+lower topology, hull-kernel topology, Jacobson topology, structure topology, primitive spectrum
+
+-/
+
+variable {α}
+
+open TopologicalSpace
+open Topology
+open Set
+open Set.Notation
+
+section SemilatticeInf
+
+variable [SemilatticeInf α]
+namespace PrimitiveSpectrum
+
+/-- For `a` of type `α` the set of element of `T` which dominate `a` is the `hull` of `a` in `T`. -/
+abbrev hull (T : Set α) (a : α) := T ↓∩ Ici a
+
+variable {T : Set α}
+
+/- The set of relative-closed sets of the form `hull T a` for some `a` in `α` is closed under
+pairwise union. -/
+lemma hull_union_hull (hT : ∀ p ∈ T, InfPrime p) (a b : α) :
+    hull T a ∪ hull T b = hull T (a ⊓ b) := by
+  ext p
+  constructor <;> intro h
+  · rcases h with (h1 | h3)
+    · exact inf_le_of_left_le h1
+    · exact inf_le_of_right_le h3
+  · exact (hT p p.2).2 h
+
+/- The set of relative-open sets of the form `(hull T a)ᶜ` for some `a` in `α` is closed under
+pairwise intersection. -/
+lemma hull_compl_inter_hull_compl (hT : ∀ p ∈ T, InfPrime p) (a b : α) :
+    (hull T a)ᶜ ∩ (hull T b)ᶜ = (hull T (a ⊓ b))ᶜ := by
+  rw [← (hull_union_hull hT), compl_union]
```

Comment:
This lemma looks a bit weird and is a direct consequence of `hull_union_hull`. I would drop it